### PR TITLE
GH-93207: Remove configure check for stdarg.h (HAVE_STDARG_PROTOTYPES)

### LINKED ISSUE #93207: Remove check for stdarg.h (HAVE_STDARG_PROTOTYPES) in configure.ac
CPython checks for `stdarg.h` as `HAVE_STDARG_PROTOTYPES` [^1] at some places [^2] but not at other places [^3]. Either this check should be repeated at every call site or should be removed all together, most compilers supports this nowadays (should be verified with buildbots). 

[^1]: https://github.com/python/cpython/blob/16a7e4a0b75080275bf12cfb71d54b01d85099b2/configure.ac#L5026

[^2]:https://github.com/python/cpython/blob/16a7e4a0b75080275bf12cfb71d54b01d85099b2/Modules/_asynciomodule.c#L2646

[^3]: https://github.com/python/cpython/blob/16a7e4a0b75080275bf12cfb71d54b01d85099b2/Objects/call.c#L557

cc @vstinner @tiran @erlend-aasland

### PR DESCRIPTION
**Pull request by kumaraditya303**
Closes #93207

### CODE REVIEW

**Comment by erlend-aasland** [comment]
AC changes look good, but I would try to investigate further which compilers/build environments this will affect (if any).

**Comment by kumaraditya303** [comment]
> AC changes look good, but I would try to investigate further which compilers/build environments this will affect (if any); buildbots do not cover everything.

It is unlikely as there is already code which uses variadic args and does not checks for `HAVE_STDARG_PROTOTYPES` and is working.

**Comment by kumaraditya303** [comment]
@erlend-aasland: `va_start` macro requires two arguments and if you try to replace `va_start(vargs, format);` with `va_start(vargs);` then it won't even compile. I tried it and got this compilation error proving that `va_start` with one argument is wrong. Hence this PR is correct and fixes the issue.

```console
/workspaces/cpython/Modules/_asynciomodule.c:2646:19: error: macro "va_start" requires 2 arguments, but only 1 given
 2646 |     va_start(vargs);
      |                   ^
```

File: Misc/NEWS.d/next/Build/2022-05-25-13-56-00.gh-issue-93207.B9Rubf.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove configure check for ``stdarg.h`` as it was not required. Patch by Kumar Aditya.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Build/2022-05-25-13-56-00.gh-issue-93207.B9Rubf.rst]: I suggest to rephrase it differently:

va_start() with two parameters, like ``va_start(args, format)``, is now required to build Python. va_start() is no longer called with a single parameter.

I suggest to copy this entry to the What's New in Python 3.12 document, in the Build Changes.

Comment (kumaraditya303) [reply Misc/NEWS.d/next/Build/2022-05-25-13-56-00.gh-issue-93207.B9Rubf.rst]: done.

**Review by erlend-aasland — APPROVED** [review]

**Comment by vstinner** [comment]
Many files already call va_start() with 2 arguments: Objects/call.c, PC/launcher.c and PC/launcher2.c, Parser/pegen_errors.c, etc.

**Comment by kumaraditya303** [comment]
> Many files already call va_start() with 2 arguments: Objects/call.c, PC/launcher.c and PC/launcher2.c, Parser/pegen_errors.c, etc. (Without testing if the HAVE_STDARG_PROTOTYPES macro is defined.)

Done

**Comment by vstinner** [comment]
Merged, thanks.

**Comment by tacaswell** [comment]
This broke cython which appears to have been relying on HAVE_STDARG_PROTOTYPES to be defined by CPython.  I am going to open a PR an the cython side to always go with the 2 argument version.

**Pull request merged**